### PR TITLE
Enh: sharing wrappers

### PR DESCRIPTION
**Pull request by feefladder**
Nobody responded to #17 so I just made my contribution.

Basically, I created wrapper functions for all API calls described in the [sharing API section](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-share-api.html#)

I added one test, since I didn't know if this is all wanted.

kind regards :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +28,11 @@
     "@babel/preset-env": "^7.8.7",
     "@babel/preset-typescript": "^7.8.3",
     "@nextcloud/browserslist-config": "^2.1.0",
-    "typedoc": "^0.20.35",
+    "@types/jest": "^29.5.5",
+    "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
+    "ts-node": "^10.9.1",
+    "typedoc": "^0.25.1",
```
Comment (feefladder) [inline package.json:35]: by `npm audit --fix --force`

File: package.json [hunk 2]
```diff
@@ -26,7 +28,11 @@
     "@babel/preset-env": "^7.8.7",
     "@babel/preset-typescript": "^7.8.3",
     "@nextcloud/browserslist-config": "^2.1.0",
-    "typedoc": "^0.20.35",
+    "@types/jest": "^29.5.5",
+    "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
+    "ts-node": "^10.9.1",
```
Comment (feefladder) [inline package.json:34]: For testing

File: package.json [hunk 3]
```diff
@@ -18,6 +18,8 @@
   "author": "Christoph Wurst",
   "license": "GPL-3.0-or-later",
   "dependencies": {
+    "@nextcloud/axios": "^2.4.0",
+    "@nextcloud/router": "^2.1.2",
```
Comment (feefladder) [inline package.json:22]: for wrapper functions

File: tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "compilerOptions": {
-        "target": "es5",
+        "target": "ES6",
```
Comment (feefladder) [inline tsconfig.json:3]: Was needed for building documentation, due to the following error:
```
> @nextcloud/sharing@0.1.0 build:doc
> typedoc --out dist/doc lib && touch dist/doc/.nojekyll

node_modules/@jest/reporters/build/index.d.ts:136:3 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

136   #private;
      ~~~~~~~~

node_modules/jest-runner/build/index.d.ts:101:3 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

101   #private;
```

File: lib/index.ts [hunk 5]
```diff
@@ -1,4 +1,9 @@
-export enum Type {
+import { generateOcsUrl } from "@nextcloud/router";
+import axios from '@nextcloud/axios'
+
+import type { AxiosResponse } from 'axios'
+
+export enum NcShareType {
```
Comment (feefladder) [inline lib/index.ts:6]: To prevent naming conflicts in libraries that use it

File: lib/index.ts [hunk 6]
```diff
@@ -10,3 +15,290 @@ export enum Type {
...
+            statuscode: number,
+        }
+    }
+}
+
+/**
+ * interface, generated using QuickType from response object
+ * TODO: find out where this is defined and make better
+ */
+export interface ShareData {
```
Comment (feefladder) [inline lib/index.ts:37]: maybe name Nc(OCS)ShareDATA?

File: jest.config.ts
Comment (feefladder) [inline jest.config.ts:1]: Autogenerated by `npm run test -- --init`

**Comment by feefladder** [comment]
Nothing is outdated, I just signed off the commits

File: lib/index.ts [hunk 7]
```diff
@@ -10,3 +15,290 @@ export enum Type {
     SHARE_TYPE_ROOM = 10,
     SHARE_TYPE_DECK = 12,
 }
+
+/**
+ * Successful OCS responses look like this, `T` is a template for the response data inside the response
+ */
+export interface NcOCSResponse<T>{
```
Comment (susnux) [inline lib/index.ts:22]: This is already provided by `@nextcloud/typings`

File: lib/index.ts (hunk 5 above)
Comment (susnux) [reply lib/index.ts:6]: Agree that `Type` is very generic, but `ShareType` should be good.
If it conflicts then just import it with alias like `import { ShareType as NcShareType } from '...'`.

Especially as we do not use prefixes on e.g. nextcloud-files or all other libraries

File: lib/index.ts (hunk 6 above)
Comment (susnux) [inline lib/index.ts:37]: No sure about this, maybe instead like the `Folder` or `Node` class of nextcloud-files?

**Comment by feefladder** [comment]
Oh wow, I'm like completely out of this and would need quite some reading/comprehending to get back to knowing what is going on... Sorry folks, but will close for now. If someone else feels the need to develop this further, I think they can build on top of this...

**Pull request closed without merging**